module-4/longestString.js

```diff
@@ -5,4 +5,32 @@
  *
  * @param {Array.<string>} strings
  * @returns {string} longest string or empty string in other cases
- */
\ No newline at end of file
+ */
+function longestString (strings){
+    let longest = "";
+        
+    if (Array.isArray(strings)) {
+        let newArray = strings.filter(el => typeof el === 'string' || Array.isArray(el));
+            
+            for (let i in newArray){
+                if (Array.isArray(newArray[i])){
+                   longestString(newArray[i])
+                }
+                
+                if (longest.length < newArray[i].length){
+                    longest = newArray[i]
+                } else if (longest.length === newArray[i].length){
+                    let sortArray = [longest, newArray[i]];
+                    longest = sortArray.sort()[0];
+                } else {
+                    longest = longest
+                }
+            }
+    } else {
+        longest = ""
```

Comment:
At this point longest is already an empty string

Reply:
done